library/lvm_gensym.py

```diff
@@ -0,0 +1,165 @@
+from ansible.module_utils.basic import AnsibleModule 
+from ansible.module_utils import facts
+
+ANSIBLE_METADATA = {
+    'metadata_version': '1.1',
+    'status': ['preview'],
+    'supported_by': 'community'
+}
+
+DOCUMENTATION = '''
+---
+module: lvm_gensym
+short_description: Generate default names for lvm variables
+version_added: "2.4"
+description: 
+    - "Module accepts two input strings consisting of a file system type and 
+       a mount point path, and outputs names based on system information"
+options:
+    fs_type:
+        description:
+            - String describing the desired file system type 
+        required: true 
+    mount:
+        description:
+            - String describing the mount point path 
+        required: true
+author: 
+    - Tim Flannagan (tflannag@redhat.com)
+'''
+
+EXAMPLES = '''
+- name: Generate names 
+  lvm_gensym:
+    fs_type: "{{ fs_type }}"
+    mount: "{{ mount_point }}"
+  register: lvm_results
+  when: lvm_vg == "" && mount_point != "" and fs_type != ""
+'''
+
+RETURN = '''
+vg_name:
+    description: The default generated name for an unspecified volume group
+    type: str 
+'''
+
+RETURN = '''
+vg_name:
+    description: The default generated name for an unspecified volume group
+    type: str 
+
+lv_name:
+    description: The default generated name for an unspecified logical volume 
+    type: str
+'''
+
+def is_unique(name, used_names):
+    if name not in used_names:
+        return True 
+    
+    return False 
+
+def init_vg_name(host_name, vg_names):
+    def get_os_name():    
+        for line in open('/etc/os-release').readlines():
+            if not line.find('ID='):
+                name = line[3:]
+            
+        name = name.replace('\n', '').replace('"', '')
+        return name 
+
+    # Initialize necessary variables
+    os_name = get_os_name()
+
+    if host_name != None and len(host_name) != 0:
+        vg_default = os_name + '_' + host_name
+    else:
+        vg_default = os_name + '_default' 
+
+    return vg_default
+
+def get_vg_name(host_name, vg_names):
+    name = init_vg_name(host_name, vg_names)
+    counter = 0
+    unique_name = is_unique(name, vg_names)
+
+    while not unique_name:
+        if counter == 0:
+            name = name + '_' + str(counter)
+        else:
+            name = name[:-2] + '_' + str(counter)
+        counter += 1
+        unique_name = is_unique(name, vg_names) 
+
+    vg_names.append(name)
+    return name 
+
+# @param_list = [ fs_type, mount ] both type str 
+def init_lv_name(param_list, lv_names):
+    # Generate name based on supplied input params
+    if 'swap' in param_list[0].lower():
+        lv_default = 'swap'
+    elif param_list[1].startswith('/'):
+        if param_list[1] == '/':
+            lv_default = 'root'
+        else:
+            lv_default = param_list[1][1:].replace('/', '_')
+    else:
+        lv_default = 'mount_default'
+
+    return lv_default 
+    
+def get_lv_name(param_list, lv_names):
```

Comment:
I already mentioned this below, but this would be easier to read/understand if you passed in the params with meaningful names instead of hardcoded indexing into a list.